packages/block-library/src/navigation/edit/navigation-menu-selector.js

```diff
@@ -173,27 +173,22 @@ function NavigationMenuSelector( {
 							} ) }
 						</MenuGroup>
 					) }
-
-					{ canUserCreateNavigationMenu && (
-						<MenuGroup label={ __( 'Tools' ) }>
-							<MenuItem
-								disabled={ isCreatingMenu }
-								onClick={ () => {
-									onClose();
-									onCreateNew();
-									setIsCreatingMenu( true );
-								} }
-							>
-								{ __( 'Create new menu' ) }
-							</MenuItem>
-						</MenuGroup>
-					) }
 				</>
 			) }
 		</DropdownMenu>
 	);
 
-	return NavigationMenuSelectorDropdown;
+	// TODO If there are more than one...
+	if (
+		( showNavigationMenus &&
+			hasNavigationMenus &&
+			navigationMenus?.length > 1 ) ||
+		( showClassicMenus && hasClassicMenus )
+	) {
+		return NavigationMenuSelectorDropdown;
+	}
```

Comment:
Let's explain what's happening here with the conditional rendering. I think it might be more readable using a tenary and having the conditionals extracted to a named `const`.